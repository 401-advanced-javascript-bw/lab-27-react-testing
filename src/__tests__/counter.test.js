import React from 'react';
import Counter from '../components/counter/counter.js';
import renderer from 'react-test-renderer'; //used for snapshot testing

describe('<Counter/>', () => {
  test('basic rendering', () => {
    const mountedCounter = shallow(<Counter />);
    expect(mountedCounter.find('span')).toBeTruthy();
  });
  test('testing state changes', () => {
    const mountedCounter = mount(<Counter />);
    const button = mountedCounter.find('a.up');
    button.simulate('click');
    expect(mountedCounter.state('count')).toEqual(1);
  });
  test('rendering follows snapshots', () => {
    const snapshot = renderer.create(<Counter />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
