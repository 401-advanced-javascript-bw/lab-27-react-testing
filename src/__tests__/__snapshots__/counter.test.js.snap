// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> rendering follows snapshots 1`] = `
<section
  className="counter"
>
  <a
    className="up clicker"
    href="/#"
    onClick={[Function]}
  >
    +
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="down clicker"
    href="/#"
    onClick={[Function]}
  >
    -
  </a>
</section>
`;
